blocks/bricks/conv.py

```diff
@@ -335,9 +335,15 @@ def get_dim(self, name):
         if name == 'input_':
             return self.input_dim
         if name == 'output':
-            return tuple(Pool.out_shape(
-                self.input_dim, self.pooling_size, st=self.step,
-                ignore_border=self.ignore_border, padding=self.padding))
+            if None in self.input_dim[-2:]:
```

Comment:
What happens to subsequent layers if they get only one None in their output_dim though? The thing that concerns me is that the non-None value will be incorrect.